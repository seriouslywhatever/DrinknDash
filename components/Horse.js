import React from 'react';
import Animated, { withTiming, useAnimatedStyle, Easing } from "react-native-reanimated";

const jockey = require('../assets/jockey.gif');

/**
 * Horse component to move in Game screen.
 * Retrieve values to determine image size and movement distance.
 */
export default Horse = (props) => {

    const translateX = props.distance;
    const size = props.size;

    const animatedStyles = useAnimatedStyle(() => ({
        transform: [{
            translateX: withTiming(translateX.value * 2, {
                duration: 750,
                easing: Easing.ease,
            })
        }],
    }));

    return (
        <Animated.View style={[animatedStyles, { width: size, height: size }]}>
            <Animated.Image
                source={jockey}
                style={{
                    width: size,
                    height: size,
                    resizeMode: 'contain',
                }}
            />
        </Animated.View>
    );
}