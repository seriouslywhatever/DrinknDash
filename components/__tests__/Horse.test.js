import React from 'react';
import renderer from 'react-test-renderer';
import Horse from '../Horse'; 

describe('Horse', () => {
    it('renders correctly', () => {
        const distance = { value: 100 };
        const size = 50; 

        const tree = renderer
            .create(<Horse distance={distance} size={size} />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

});
