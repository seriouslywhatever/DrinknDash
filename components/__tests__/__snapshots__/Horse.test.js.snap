// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Horse renders correctly 1`] = `
<View
  animatedStyle={
    {
      "value": {
        "transform": [
          {
            "translateX": 200,
          },
        ],
      },
    }
  }
  collapsable={false}
  style={
    {
      "height": 50,
      "transform": [
        {
          "translateX": 200,
        },
      ],
      "width": 50,
    }
  }
>
  <Image
    animatedStyle={
      {
        "value": {},
      }
    }
    collapsable={false}
    source={1}
    style={
      {
        "height": 50,
        "resizeMode": "contain",
        "width": 50,
      }
    }
  />
</View>
`;

exports[`Horse renders correctly with given props 1`] = `
<View
  animatedStyle={
    {
      "value": {
        "transform": [
          {
            "translateX": 200,
          },
        ],
      },
    }
  }
  collapsable={false}
  style={
    {
      "height": 50,
      "transform": [
        {
          "translateX": 200,
        },
      ],
      "width": 50,
    }
  }
>
  <Image
    animatedStyle={
      {
        "value": {},
      }
    }
    collapsable={false}
    source={1}
    style={
      {
        "height": 50,
        "resizeMode": "contain",
        "width": 50,
      }
    }
  />
</View>
`;
